core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -375,6 +375,9 @@ wasi_fd_pread(wasm_exec_env_t exec_env, wasi_fd_t fd, iovec_app_t *iovec_app,
         return (wasi_errno_t)-1;
 
     total_size = sizeof(wasi_iovec_t) * (uint64)iovs_len;
+    if (total_size == 0) {
```

Comment:
if `total_size` is zero (i.e. `iovs_len` is zero), can't we just return early, instead of doing the rest of the work in this function?

Reply:
maybe we can.
but i don't think it makes sense to optimize this rarely-used case.
it's better to keep it consistent with the normal case. (eg. validate fd etc)